homeassistant/components/icloud/account.py

```diff
@@ -110,9 +115,36 @@ def setup(self) -> None:
                 self._icloud_dir.path,
                 with_family=self._with_family,
             )
-        except PyiCloudFailedLoginException as error:
+        except PyiCloudFailedLoginException:
```

Comment:
You should also catch this into the `keep_alive` function (try-catch between `self.api.authenticate()`) so when the integration tries to update and has a login issue, it asked to reconnect.

Use case:
- start HA
- change password
- updates will fail --> should ask to log agin.

Add your re_auth process into one function to avoid duplication.

Reply:
I don't think an exception is getting raised during the `keep_alive`. I am getting the notifications to refresh 2FA on my phone but I don't see any errors in the HA logs, so there is nothing to catch unless I am misunderstanding your proposal